mllib/src/test/scala/org/apache/spark/ml/regression/LinearRegressionSuite.scala

```diff
@@ -93,525 +96,641 @@ class LinearRegressionSuite extends SparkFunSuite with MLlibTestSparkContext {
+         as.numeric.data3.V2. 4.70011
+         as.numeric.data3.V3. 7.19943
+       */
+      val weightsWithoutInterceptR = Vectors.dense(4.70011, 7.19943)
+
+      assert(modelWithoutIntercept1.intercept ~== 0 absTol 1E-3)
+      assert(modelWithoutIntercept1.weights ~= weightsWithoutInterceptR relTol 1E-3)
+      assert(modelWithoutIntercept2.intercept ~== 0 absTol 1E-3)
+      assert(modelWithoutIntercept2.weights ~= weightsWithoutInterceptR relTol 1E-3)
+    }
   }
 
   test("linear regression with intercept with L1 regularization") {
-    val trainer1 = (new LinearRegression).setElasticNetParam(1.0).setRegParam(0.57)
-    val trainer2 = (new LinearRegression).setElasticNetParam(1.0).setRegParam(0.57)
-      .setStandardization(false)
-    val model1 = trainer1.fit(dataset)
-    val model2 = trainer2.fit(dataset)
-
-    /*
-       weights <- coef(glmnet(features, label, family="gaussian", alpha = 1.0, lambda = 0.57))
-       > weights
-        3 x 1 sparse Matrix of class "dgCMatrix"
-                                 s0
-       (Intercept)         6.24300
-       as.numeric.data.V2. 4.024821
-       as.numeric.data.V3. 6.679841
-     */
-    val interceptR1 = 6.24300
-    val weightsR1 = Vectors.dense(4.024821, 6.679841)
-
-    assert(model1.intercept ~== interceptR1 relTol 1E-3)
-    assert(model1.weights ~= weightsR1 relTol 1E-3)
-
-    /*
-      weights <- coef(glmnet(features, label, family="gaussian", alpha = 1.0, lambda = 0.57,
-        standardize=FALSE))
-      > weights
-       3 x 1 sparse Matrix of class "dgCMatrix"
-                                s0
-      (Intercept)         6.416948
-      as.numeric.data.V2. 3.893869
-      as.numeric.data.V3. 6.724286
-     */
-    val interceptR2 = 6.416948
-    val weightsR2 = Vectors.dense(3.893869, 6.724286)
-
-    assert(model2.intercept ~== interceptR2 relTol 1E-3)
-    assert(model2.weights ~= weightsR2 relTol 1E-3)
-
-
-    model1.transform(dataset).select("features", "prediction").collect().foreach {
-      case Row(features: DenseVector, prediction1: Double) =>
-        val prediction2 =
-          features(0) * model1.weights(0) + features(1) * model1.weights(1) + model1.intercept
-        assert(prediction1 ~== prediction2 relTol 1E-5)
+    breakable {
+      Seq("auto", "l-bfgs", "normal").foreach { solver =>
+        val trainer1 = (new LinearRegression).setElasticNetParam(1.0).setRegParam(0.57)
+          .setSolver(solver)
+        val trainer2 = (new LinearRegression).setElasticNetParam(1.0).setRegParam(0.57)
+          .setSolver(solver).setStandardization(false)
+
+        var model1: LinearRegressionModel = null
+        var model2: LinearRegressionModel = null
+
+        // Normal optimizer is not supported with only L1 regularization case.
+        if (solver == "normal") {
+          intercept[IllegalArgumentException] {
+            trainer1.fit(dataset)
+            trainer2.fit(dataset)
+          }
+          break
+        } else {
```

Comment:
break will break the entire loop, so it depends on the ordering of Seq, and it only works for the last element in the Seq.  This will easily cause some bugs.

See,
```scala
scala> :paste
// Entering paste mode (ctrl-D to finish)


import scala.util.control.Breaks.break
import scala.util.control.Breaks.breakable
breakable {
Seq("auto", "l-bfgs", "normal").foreach { solver =>
   println(s"starting: $solver")

   if (solver == "l-bfgs") {
     break
   }
   println(s"ending: $solver")
}
}


// Exiting paste mode, now interpreting.

starting: auto
ending: auto
starting: l-bfgs
```

I suggest that move the whole test code into else statement,

```scala
} else {
var model1 = trainer1.fit(dataset)
val model2 = trainer2.fit(dataset)
  val interceptR1 = 6.24300
  val weightsR1 = Vectors.dense(4.024821, 6.679841)
  assert(model1.intercept ~== interceptR1 relTol 1E-3)
  assert(model1.weights ~= weightsR1 relTol 1E-3)
  ...
  ...
}
```

Also, fix the indentation in the comment.